tests/python_package_test/test_engine.py

```diff
@@ -970,6 +970,29 @@ def test_continue_train_multiclass():
     assert evals_result['valid_0']['multi_logloss'][-1] == pytest.approx(ret)
 
 
+def test_continue_train_different_feature_size(capsys):
+    np.random.seed(0)
+    train_X = np.hstack([np.ones(800).reshape(-1, 8), np.arange(200, 0, -1).reshape(-1, 2)])
+    train_y = np.sum(train_X[:, -2:], axis=1)
+    train_data = lgb.Dataset(train_X, label=train_y)
+    params = {
+        "objective": "regression",
+        "num_trees": 10,
+        "num_leaves": 31,
+        "verbose": -1,
+        'predict_disable_shape_check': True,
```

Comment:
Ah, good observation! But `predict()` is called internally here when you set init scores by a previously trained model
https://github.com/microsoft/LightGBM/blob/0018206742544cfd0b2c1e156f177ab5f468a998/python-package/lightgbm/engine.py#L153-L154
https://github.com/microsoft/LightGBM/blob/0018206742544cfd0b2c1e156f177ab5f468a998/python-package/lightgbm/engine.py#L160-L163
https://github.com/microsoft/LightGBM/blob/0018206742544cfd0b2c1e156f177ab5f468a998/python-package/lightgbm/basic.py#L1393
https://github.com/microsoft/LightGBM/blob/0018206742544cfd0b2c1e156f177ab5f468a998/python-package/lightgbm/basic.py#L1399-L1402

Reply:
Ah, I see! So then is the bug addressed by this PR only present when you've also set `predict_disable_shape_check = True`? I expect that most LightGBM users will not be aware that that prediction parameter is relevant for training continuation.

Reply:
Yes, without setting `predict_disable_shape_check = True` users get the following error:
```
LightGBMError: The number of features in data (5) is not the same as it was in training data (10).
You can set ``predict_disable_shape_check=true`` to discard this error, but please be aware what you are doing.
```
I think the second sentence of the error "You can set ``predict_disable_shape_check=true`` to discard this error ..." helps understand this complicated relationship.

Reply:
`predict_disable_shape_check ` is necessary because here a numpy matrix is used in the test case. If we use a LibSVM file as dataset in the test case, we don't need `predict_disable_shape_check=True`. I use a numpy matrix here because I think it is more convenient than LivSVM in that it does not requiring generating files in test cases.